core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushJoinIntoTableScan.java

```diff
@@ -114,6 +115,19 @@ public Result apply(JoinNode joinNode, Captures captures, Context context)
             return Result.empty();
         }
 
+        if (getJoinPushdownMode(context.getSession()) == JoinPushdownMode.AUTOMATIC) {
```

Comment:
ideally use a switch, and make it exhaustive, future proofing for the case when we add something like AUTOMATIC_EAGER (which we don't have to add yet, but we may want to add in the future)

Reply:
nvrm, in this case it doesn't matter -- this is the only place the enum is used, so no way it gets forgotten and not updated